application/src/main/java/org/thingsboard/server/service/subscription/DefaultTbLocalSubscriptionService.java

```diff
@@ -143,7 +143,20 @@ public void onApplicationEvent(ClusterTopologyChangeEvent event) {
              * Even if we cache locally the list of active subscriptions by entity id, it is still time-consuming operation to get them from cache
              * Since number of subscriptions is usually much less than number of devices that are pushing data.
              */
-            subscriptionsByEntityId.values().forEach(sub -> pushSubEventToManagerService(sub.getTenantId(), sub.getEntityId(), sub.toEvent(ComponentLifecycleEvent.UPDATED)));
+            Set<UUID> staleSubs = new HashSet<>();
+            subscriptionsByEntityId.forEach((id, sub) -> {
+                try {
+                    pushSubEventToManagerService(sub.getTenantId(), sub.getEntityId(), sub.toEvent(ComponentLifecycleEvent.UPDATED));
+                } catch (TenantNotFoundException e) {
+                    staleSubs.add(id);
+                    log.warn("Cleaning up stale subscription {} for tenant {} due to TenantNotFoundException", id, sub.getTenantId());
+                } catch (Exception e) {
+                    log.error("Failed to push subscription {} to manager service", sub, e);
+                }
+            });
+            if (!staleSubs.isEmpty()) {
```

Comment:
Missing `subsLock.lock();` on removing the commands. Please add it. Also, don't forget to call `entityUpdates.remove(entityId.getId());`